File: testProjects/dcVarTest/VampBias.m
% file: VampBias.m
% SLiCAP-MATLAB script file
% You should call this script from the file that performs the 
% initialization of a project for your SLiCAP-MATLAB work.
% You need to store this file as well as the circuit file(s) 
% in the project directory.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
checkCircuit('VampBias');
htmlPage('Netlist and circuit data');
netlist2html('VampBias');
elementData2html();
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
htmlPage('DC variance analysis');
simType('symbolic');
gainType('vi');
dataType('dcvar');
detector('V_out');
dcVar2html(execute());
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
htmlPage('Biasing results');
dataType('dc');
result = execute();
DCvalue = result.results(1);
syms 'V_out' 'sigma_Vout'
head2html('The DC output voltage');
text2html('The DC output voltage is obtained as:');
eqn2html(V_out, DCvalue);
dataType('dcvar');
result = execute();
DCvarResult = result.results(1);
detRefVar = DCvarResult(5);       % Detector-referred variance
head2html('DC output voltage variance');
text2html('The variance of the DC output voltage is obtained as:');
eqn2html((sigma_Vout)^2, detRefVar);
stophtml();

File: testProjects/dcVarTest/VampBias.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
"""
Created on Sun Jul  5 08:59:59 2020

@author: anton
"""
from SLiCAP import *
t1 = time()

prj = initProject('Vamp bias project') # Creates the SLiCAP libraries and the
                             # project HTML index page

fileName = 'VampBias.cir'
i1 = instruction()           # Creates an instance of an instruction object
i1.setCircuit(fileName)      # Checks and defines the local circuit object and
                             # sets the index page to the circuit index page
i1.setDetector('V_out')
i1.setSimType('symbolic')
i1.setGainType('vi')
i1.setDataType('dcvar')
result = i1.execute()

htmlPage('Netlist and circuit data');
netlist2html(fileName);
elementData2html(i1.circuit);
htmlPage('DC variance analysis');
dcVar2html(result)
t2 = time()
print '\n', t2-t1, 's'